Fix ready() method in FileWatcher

daniel-zullo-frequenz opened the pull request:
The ready() method now returns False when an exception is set.

Fixes #317

daniel-zullo-frequenz committed:
Fix FileWatcher ready() method

The ready() method now returns False when an exception
is set.

Signed-off-by: Daniel Zullo <daniel.zullo@frequenz.com>

daniel-zullo-frequenz committed:
Add a note to the FileWatcher documentation

To clarify the responsibility of the owner of the
FileWatcher instance.

Signed-off-by: Daniel Zullo <daniel.zullo@frequenz.com>

daniel-zullo-frequenz committed:
Update release-notes with bug-fix for FileWatcher

Signed-off-by: Daniel Zullo <daniel.zullo@frequenz.com>

llucax reviewed (commented):
The diff LGTM, but the PR still needs some improvements:

* Update the PR title as this only fix a bug now
* Update the PR body as it will be part of the merge commit, it doesn't make sense to have outdated, struck out, information.
* It needs release notes

daniel-zullo-frequenz commented:
Updated PR tittle and description. The release-notes was already added, do you want me to change something there?

llucax commented:
Woops, I guess I was looking commit-by-commit and forget to hit next at some point.

llucax commented on RELEASE_NOTES.md:
```diff
@@ -14,4 +14,4 @@
 ## Bug Fixes
 
-<!-- Here goes notable bug fixes that are worth a special mention or explanation -->
+- `FileWatcher`: Fixed `ready()` method to return False when an error occurs.
```
```suggestion
- `FileWatcher`: Fixed `ready()` method to return False when an error occurs. Before this fix, `select()` (and other code using `ready()`) never detected the `FileWatcher` was stopped and the `select()` loop was continuously waking up to inform the receiver was ready.
```

llucax reviewed (dismissed):
Just an optional improvement over the release notes, trying to explain a bit more the implications of the bug, which is what users care more about (just knowing that a `return False` was missed in a method they almost never use is not that informative).

daniel-zullo-frequenz committed:
Improve release-notes entry

This was suggested in the review to extend
the information in the release-notes entry.

Co-authored-by: Leandro Lucarella <luca@llucax.com>
Signed-off-by: daniel-zullo-frequenz <120166726+daniel-zullo-frequenz@users.noreply.github.com>

daniel-zullo-frequenz commented:
> Just an optional improvement over the release notes, trying to explain a bit more the implications of the bug, which is what users care more about (just knowing that a return False was missed in a method they almost never use is not that informative).

Thanks! Added to suggestion to  the release-notes
